src/modlunit/units.cpp

```diff
@@ -109,28 +113,25 @@ static int Getc(FILE* inp) {
 #define UNIT_STK_SIZE 20
 static struct unit unit_stack[UNIT_STK_SIZE], *usp{nullptr};
 
-static char* neuronhome() {
+static std::string hoc_dos2unixpath(std::string d) {
+    std::replace(d.begin(), d.end(), '\\', '/');
+    return d;
+}
+
+static std::string neuronhome() {
 #if defined(MINGW)
-    int i;
-    static char buf[256];
-    GetModuleFileName(NULL, buf, 256);
-    for (i = strlen(buf); i >= 0 && buf[i] != '\\'; --i) {
-        ;
-    }
-    buf[i] = '\0';  // /neuron.exe gone
-    //      printf("setneuronhome |%s|\n", buf);
-    for (i = strlen(buf); i >= 0 && buf[i] != '\\'; --i) {
-        ;
-    }
-    buf[i] = '\0';  // /bin gone
-    {
-        char* u = hoc_dos2unixpath(buf);
-        strcpy(buf, hoc_dos2unixpath(u));
-        free(u);
-    }
+    std::string buf(256, '\0');
+    GetModuleFileName(nullptr, buf.data(), 256);
+    // Remove neuron.exe
+    auto pos = buf.find_last_of('\\');
+    buf.resize(pos);
+    // Remove bin
+    pos = buf.find_last_of('\\');
+    buf.resize(pos);
+    std::replace(buf.begin(), buf.end(), '\\', '/');
     return buf;
 #else
-    return getenv("NEURONHOME");
+    return std::getenv("NEURONHOME");
```

Comment:
Since `getenv` returns a `char*` and not a `std::string`, do we get undefined behavior if `NEURONHOME` is not set?

Reply:
It will only return an empty `std::string`, right?